### PR TITLE
send notifications from rawls

### PR DESCRIPTION
**Pull request by dvoet**
https://broadinstitute.atlassian.net/browse/GAWB-1652

also review
https://github.com/broadinstitute/thurloe/pull/40
https://github.com/broadinstitute/firecloud-orchestration/pull/390
https://github.com/broadinstitute/firecloud-develop/pull/330 (local)
https://github.com/broadinstitute/firecloud-develop/pull/331 (dev)
https://github.com/broadinstitute/firecloud-develop/pull/332 (fiab)

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Check that the **Product Owner** has signed off on any user-facing changes
- [x] **Submitter**: Make sure Swagger is updated if API changes
  - [ ] **...and Orchestration's Swagger too!**
- [x] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Database checks:
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] Tell your tech lead (TL) that the PR exists if they want to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [x] **TL** sign off
- [x] **LR** sign off
- [x] **Assign to submitter** to finalize
- [x] **Submitter**: Verify all tests go green, including CI tests
- [x] **Submitter**: Squash commits and merge to develop
- [x] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala [hunk 1]
```diff
@@ -503,6 +506,10 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
       dataSource.inTransaction { dataAccess =>
         withWorkspaceContext(workspaceName, dataAccess) { workspaceContext =>
           val dedupedInvites = invites.filterNot(update => existingInvites.contains(WorkspaceACLUpdate(update.email, update.accessLevel, None)))
+
+          val invitedNotifications = dedupedInvites.map(invite => Notifications.WorkspaceInvitedNotification(invite.email, userInfo.userEmail))
```
Comment (MatthewBemis) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala]: i think we need to do another round of filtering here before we actually get to the list of invite emails that we want to send. basically, if someone updates the `accessLevel` of a pending invite, we don't want to send them another invite to FireCloud

`dedupedInvites` filters out any invites that are identical to ones that already exist, but we still need to do something like: https://github.com/broadinstitute/rawls/pull/638/files#diff-91fe8efb10dfe91234a26e4f80039fc6R525

Comment (MatthewBemis) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala]: i think we need to do another round of filtering here before we actually get to the list of invite emails that we want to send. basically, if someone updates the `accessLevel` of a pending invite, we don't want to send them another invite to FireCloud

`dedupedInvites` filters out any invites that are identical to ones that already exist, but we still need to do something like: https://github.com/broadinstitute/rawls/pull/638/files#diff-91fe8efb10dfe91234a26e4f80039fc6R525

File: project/Version.scala [hunk 2]
```diff
@@ -4,8 +4,8 @@ import sbt._
 object Version {
   val baseModelVersion = "0.1"
 
-  def getVersionString(baseDir: File) = {
-    def getLastModelCommitFromGit = { s"""git log -n 1 --pretty=format:%h ${baseDir.getAbsolutePath}""" !! }
```
Comment (MatthewBemis) [inline project/Version.scala:8]: can you give a 1-2 sentence explanation to me about what these changes are and why they're needed?

**Review by MatthewBemis — CHANGES_REQUESTED** [review]

File: project/Version.scala (hunk 2 above)
Comment (dvoet) [reply project/Version.scala:8]: With the old code, each of rawls-utils, workbench-google, workbench-util would get published with different git hashes (the hash of the last change in the subdir). I found it too confusing. Now they all get pushed with the same hash.

**Review by MatthewBemis — APPROVED** [review]
:+1:

**Pull request merged**